### ISSUE TITLE
Java - UseLambdaForFunctionalInterface - remove imports after

### ISSUE THREAD

**Issue by nmck257** [openrewrite/rewrite#1844]
Test case:

```kotlin
    @Test
    fun useLambdaThenRemoveUnusedImports(jp: JavaParser) = assertChanged(
        jp,
        before = """
        import java.util.HashMap;
        import java.util.function.Function;

        public class Temp {
            public static void foo(){
                new HashMap<Integer, String>().computeIfAbsent(3, new Function<Integer, String>() {
                    @Override
                    public String apply(Integer integer) {
                        return String.valueOf(integer + 1);
                    }
                });
            }
        }
        """,
        after = """
        import java.util.HashMap;

        public class Temp {
            public static void foo(){
                new HashMap<Integer, String>().computeIfAbsent(3, integer -> String.valueOf(integer + 1));
            }
        }
        """
    )
```

If a type is only used as an anonymous class, then after replacing with a lambda, the corresponding import may no longer be necessary.

**Comment by nmck257** [comment]
FYI, I'm actually planning to add this myself, as a (relatively) minimal first contribution -- just wanted to get visibility first (and an issue #)

**Comment by tkvangorder** [comment]
This was fixed with PR #1851
